source/agora/common/Transaction.d

```diff
@@ -257,37 +257,23 @@ public bool verifyData (Transaction tx,
         if (output.value < 0)
             return false;
 
-    return tx.getSumOutput() <= tx.getSumInput(findOutput);
-}
-
-/*******************************************************************************
-
-    Get result of signature verification
-
-    Params:
-        tx = `Transaction` to verify the signature of
-        findOutput = delegate for finding `Output`
+    long sum_unspent;
 
-    Return:
-        Return true if this signature is verified.
-
-*******************************************************************************/
-
-public bool verifySignature (Transaction tx,
-    Output* delegate (Hash hash, size_t index) @safe findOutput) @safe
-{
-    const Hash txHash = hashFull(tx);
+    const tx_hash = hashFull(tx);
     foreach (input; tx.inputs)
     {
-        if (auto output = findOutput(input.previous, input.index))
-        {
-            if (!output.address.verify(input.signature, txHash[]))
-                return false;
-        }
-        else
+        // all referenced outputs must be present
+        auto output = findOutput(input.previous, input.index);
+        if (output is null)
+            return false;
+
+        if (!output.address.verify(input.signature, tx_hash[]))
             return false;
+
+        sum_unspent += output.value;
```

Comment:
Hello overflow. Let's go with this right now because we don't have a type for monetary value, but it's `#2` on my priority list.